docs/reference/query-languages/esql/esql-lookup-join.md

```diff
@@ -0,0 +1,128 @@
+
+## How the `LOOKUP JOIN` command works [esql-how-lookup-join-works]
+
+The `LOOKUP JOIN` command adds new columns to a table, with data from {{es}} indices.
+
+:::{image} ../../../images/esql-lookup-join.png
+:alt: esql lookup join
+:::
+
+`<lookup_index>`
+: The name of the lookup index. This must be a specific index name - wildcards, aliases, and remote cluster references are not supported.
+
+`<field_name>`
+: The field to join on. This field must exist in both your current query results and in the lookup index. If the field contains multi-valued entries, those entries will not match anything (the added fields will contain `null` for those rows).
+
+## Example
+
+`LOOKUP JOIN` has left-join behavior. If no rows match in the looked index, `LOOKUP JOIN` retains the incoming row and adds `null`s. If many rows in the lookedup index match, `LOOKUP JOIN` adds one row per match.
+
+In this example, we have two sample tables:
+
+**employees**
+
+| birth_date|emp_no|first_name|gender|hire_date|language|
+|---|---|---|---|---|---|
+|1955-10-04T00:00:00Z|10091|Amabile    |M|1992-11-18T00:00:00Z|3|
+|1964-10-18T00:00:00Z|10092|Valdiodio  |F|1989-09-22T00:00:00Z|1|
+|1964-06-11T00:00:00Z|10093|Sailaja    |M|1996-11-05T00:00:00Z|3|
+|1957-05-25T00:00:00Z|10094|Arumugam   |F|1987-04-18T00:00:00Z|5|
+|1965-01-03T00:00:00Z|10095|Hilari     |M|1986-07-15T00:00:00Z|4|
+
+**languages_non_unique_key**
+
+|language_code|language_name|country|
+|---|---|---|
+|1|English|Canada|
+|1|English|
+|1||United Kingdom|
+|1|English|United States of America|
+|2|German|[Germany\|Austria]|
+|2|German|Switzerland|
+|2|German|
+|4|Quenya|
+|5||Atlantis|
+|[6\|7]|Mv-Lang|Mv-Land|
+|[7\|8]|Mv-Lang2|Mv-Land2|
+|Null-Lang|Null-Land|
+|Null-Lang2|Null-Land2|
+
+Running the following query would provide the results shown below.
+
+```esql
+FROM employees
+| EVAL language_code = emp_no % 10
+| LOOKUP JOIN languages_lookup_non_unique_key ON language_code
+| WHERE emp_no > 10090 AND emp_no < 10096
+| SORT emp_no, country
+| KEEP emp_no, language_code, language_name, country;
+```
+
+|emp_no|language_code|language_name|country|
+|---|---|---|---|
+|    10091      | 1                     | English               | Canada|
+|    10091      | 1                     | null                  | United Kingdom|
+|    10091      | 1                     | English               | United States of America|
+|    10091      | 1                     | English               | null|
+|    10092      | 2                     | German                | [Germany, Austria]|
+|    10092      | 2                     | German                | Switzerland|
+|    10092      | 2                     | German                | null|
+|    10093      | 3                     | null                  | null|
+|    10094      | 4                     | Quenya                | null|
+|    10095      | 5                     | null                  | Atlantis|
```

Comment:
Thank you for including the example from the docs. In a follow-up, maybe we want to construct a nicer example for the docs? (We used to place such examples in `docs.csv-spec`.) Then readers won't have to wonder why we include Atlantis or the elven language of Quenya :D

Reply:
If you think it would be cleaner, I just just replace those values with something more realistic @alex-spies

Reply:
👍 on having better sample data that ties to the existing sample data in Kibana (which can have its own separate section).